### PR TITLE
fix(cf-provider): made cloudformation provider operational

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
- bump `github.com/awslabs/goformation/v4` to `github.com/awslabs/goformation/v7`
- add example cf2tf to compare with cloudformation output
- adopt parser for cloudformation provider

Fixed: #2879

I have:

- [x] Run `make test` to ensure this PR is ready for review.

### How has this code been tested

**Cloudformation:**

```
make run ARGS="breakdown --path examples/cloudformation/cf.json --usage-file=examples/cloudformation/infracost-usage.yml"
env INFRACOST_ENV=dev go run -ldflags="-X 'github.com/infracost/infracost/internal/version.Version=v0.10.33+3-g4adec9f9'" github.com/infracost/infracost/cmd/infracost breakdown --path examples/cloudformation/cf.json --usage-file=examples/cloudformation/infracost-usage.yml
Detected CloudFormation at examples/cloudformation/cf.json
  ✔ Extracting only cost-related params from cloudformation 
  ✔ Retrieving cloud prices to calculate costs 

Project: haarchri/infracost/examples/cloudformation/cf.json

 Name                                                  Monthly Qty  Unit                  Monthly Cost 
                                                                                                       
 AWS::DynamoDB::Table.AlertsTable                                                                      
 ├─ Write request unit (WRU)                      Monthly cost depends on usage: $0.00000125 per WRUs  
 ├─ Read request unit (RRU)                       Monthly cost depends on usage: $0.00000025 per RRUs  
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 AWS::DynamoDB::Table.BusinessesTable                                                                  
 ├─ Write request unit (WRU)                      Monthly cost depends on usage: $0.00000125 per WRUs  
 ├─ Read request unit (RRU)                       Monthly cost depends on usage: $0.00000025 per RRUs  
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 AWS::DynamoDB::Table.CustomersTable                                                                   
 ├─ Write capacity unit (WCU)                                   32  WCU                         $15.18 
 ├─ Read capacity unit (RCU)                                    12  RCU                          $1.14 
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 OVERALL TOTAL                                                                                  $16.32 
──────────────────────────────────
3 cloud resources were detected:
∙ 3 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ haarchri/infracost/examples/cloudformation/cf.json ┃ $16          ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛
```

**Terraform:**

```
make run ARGS="breakdown --path examples/cf2tf --usage-file=examples/cf2tf/infracost-usage.yml"
env INFRACOST_ENV=dev go run -ldflags="-X 'github.com/infracost/infracost/internal/version.Version=v0.10.33+3-g4adec9f9'" github.com/infracost/infracost/cmd/infracost breakdown --path examples/cf2tf --usage-file=examples/cf2tf/infracost-usage.yml
Evaluating Terraform directory at examples/cf2tf
  ✔ Downloading Terraform modules 
  ✔ Evaluating Terraform directory 
  ✔ Retrieving cloud prices to calculate costs 

Project: haarchri/infracost/examples/cf2tf

 Name                                                  Monthly Qty  Unit                  Monthly Cost 
                                                                                                       
 aws_dynamodb_table.alerts_table                                                                       
 ├─ Write request unit (WRU)                      Monthly cost depends on usage: $0.00000125 per WRUs  
 ├─ Read request unit (RRU)                       Monthly cost depends on usage: $0.00000025 per RRUs  
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 aws_dynamodb_table.businesses_table                                                                   
 ├─ Write request unit (WRU)                      Monthly cost depends on usage: $0.00000125 per WRUs  
 ├─ Read request unit (RRU)                       Monthly cost depends on usage: $0.00000025 per RRUs  
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 aws_dynamodb_table.customers_table                                                                    
 ├─ Write capacity unit (WCU)                                   32  WCU                         $15.18 
 ├─ Read capacity unit (RCU)                                    12  RCU                          $1.14 
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 OVERALL TOTAL                                                                                  $16.32 
──────────────────────────────────
3 cloud resources were detected:
∙ 3 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ haarchri/infracost/examples/cf2tf                  ┃ $16          ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛
```

with usage override:
**Cloudformation:**

```
cat infracost-usage.yaml
version: 0.1
resource_usage:
  AWS::DynamoDB::Table.AlertsTable:
    monthly_write_request_units: 3000000  # Monthly write request units in (used for on-demand DynamoDB).
    monthly_read_request_units: 8000000   # Monthly read request units in (used for on-demand DynamoDB).
    storage_gb: 230                       # Total storage for tables in GB.
    pitr_backup_storage_gb: 2300          # Total storage for Point-In-Time Recovery (PITR) backups in GB.
    on_demand_backup_storage_gb: 460      # Total storage for on-demand backups in GB.
    monthly_data_restored_gb: 230         # Monthly size of restored data in GB.
    monthly_streams_read_request_units: 2 # Monthly streams read request units.
```

```
make run ARGS="breakdown --path examples/cloudformation/cf.json --usage-file=examples/cloudformation/infracost-usage.yml"
env INFRACOST_ENV=dev go run -ldflags="-X 'github.com/infracost/infracost/internal/version.Version=v0.10.33+3-g4adec9f9-dirty'" github.com/infracost/infracost/cmd/infracost breakdown --path examples/cloudformation/cf.json --usage-file=examples/cloudformation/infracost-usage.yml
Detected CloudFormation at examples/cloudformation/cf.json
  ✔ Extracting only cost-related params from cloudformation 
  ✔ Retrieving cloud prices to calculate costs 

Project: haarchri/infracost/examples/cloudformation/cf.json

 Name                                                  Monthly Qty  Unit                  Monthly Cost 
                                                                                                       
 AWS::DynamoDB::Table.AlertsTable                                                                      
 ├─ Write request unit (WRU)                             3,000,000  WRUs                         $3.75 
 ├─ Read request unit (RRU)                              8,000,000  RRUs                         $2.00 
 ├─ Data storage                                               230  GB                          $57.50 
 ├─ Point-In-Time Recovery (PITR) backup storage             2,300  GB                         $460.00 
 ├─ On-demand backup storage                                   460  GB                          $46.00 
 ├─ Table data restored                                        230  GB                          $34.50 
 └─ Streams read request unit (sRRU)                             2  sRRUs                        $0.00 
                                                                                                       
 AWS::DynamoDB::Table.BusinessesTable                                                                  
 ├─ Write request unit (WRU)                      Monthly cost depends on usage: $0.00000125 per WRUs  
 ├─ Read request unit (RRU)                       Monthly cost depends on usage: $0.00000025 per RRUs  
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 AWS::DynamoDB::Table.CustomersTable                                                                   
 ├─ Write capacity unit (WCU)                                   32  WCU                         $15.18 
 ├─ Read capacity unit (RCU)                                    12  RCU                          $1.14 
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 OVERALL TOTAL                                                                                 $620.07 
──────────────────────────────────
3 cloud resources were detected:
∙ 3 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ haarchri/infracost/examples/cloudformation/cf.json ┃ $620         ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛
```

with usage override:
**Terraform:**

```
cat infracost-usage.yaml
version: 0.1
resource_usage:
  aws_dynamodb_table.alerts_table:
    monthly_write_request_units: 3000000  # Monthly write request units in (used for on-demand DynamoDB).
    monthly_read_request_units: 8000000   # Monthly read request units in (used for on-demand DynamoDB).
    storage_gb: 230                       # Total storage for tables in GB.
    pitr_backup_storage_gb: 2300          # Total storage for Point-In-Time Recovery (PITR) backups in GB.
    on_demand_backup_storage_gb: 460      # Total storage for on-demand backups in GB.
    monthly_data_restored_gb: 230         # Monthly size of restored data in GB.
    monthly_streams_read_request_units: 2 # Monthly streams read request units.
```

```
make run ARGS="breakdown --path examples/cf2tf --usage-file=examples/cf2tf/infracost-usage.yml"
env INFRACOST_ENV=dev go run -ldflags="-X 'github.com/infracost/infracost/internal/version.Version=v0.10.33+3-g4adec9f9-dirty'" github.com/infracost/infracost/cmd/infracost breakdown --path examples/cf2tf --usage-file=examples/cf2tf/infracost-usage.yml
Evaluating Terraform directory at examples/cf2tf
  ✔ Downloading Terraform modules 
  ✔ Evaluating Terraform directory 
  ✔ Retrieving cloud prices to calculate costs 

Project: haarchri/infracost/examples/cf2tf

 Name                                                  Monthly Qty  Unit                  Monthly Cost 
                                                                                                       
 aws_dynamodb_table.alerts_table                                                                       
 ├─ Write request unit (WRU)                             3,000,000  WRUs                         $3.75 
 ├─ Read request unit (RRU)                              8,000,000  RRUs                         $2.00 
 ├─ Data storage                                               230  GB                          $57.50 
 ├─ Point-In-Time Recovery (PITR) backup storage             2,300  GB                         $460.00 
 ├─ On-demand backup storage                                   460  GB                          $46.00 
 ├─ Table data restored                                        230  GB                          $34.50 
 └─ Streams read request unit (sRRU)                             2  sRRUs                        $0.00 
                                                                                                       
 aws_dynamodb_table.businesses_table                                                                   
 ├─ Write request unit (WRU)                      Monthly cost depends on usage: $0.00000125 per WRUs  
 ├─ Read request unit (RRU)                       Monthly cost depends on usage: $0.00000025 per RRUs  
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 aws_dynamodb_table.customers_table                                                                    
 ├─ Write capacity unit (WCU)                                   32  WCU                         $15.18 
 ├─ Read capacity unit (RCU)                                    12  RCU                          $1.14 
 ├─ Data storage                                  Monthly cost depends on usage: $0.25 per GB          
 ├─ Point-In-Time Recovery (PITR) backup storage  Monthly cost depends on usage: $0.20 per GB          
 ├─ On-demand backup storage                      Monthly cost depends on usage: $0.10 per GB          
 ├─ Table data restored                           Monthly cost depends on usage: $0.15 per GB          
 └─ Streams read request unit (sRRU)              Monthly cost depends on usage: $0.0000002 per sRRUs  
                                                                                                       
 OVERALL TOTAL                                                                                 $620.07 
──────────────────────────────────
3 cloud resources were detected:
∙ 3 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ haarchri/infracost/examples/cf2tf                  ┃ $620         ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛
```

### CODE REVIEW

**Comment by aliscott** [comment]
Thanks @haarchri 🙏. We're not concentrating on the CloudFormation implementation just now but we'll review this so it gets into the next release.

**Comment by bdekonin** [comment]
I would also like to contribute to support other Cloudformation resources.

**Comment by aliscott** [comment]
> I would also like to contribute to support other Cloudformation resources. @aliscott Is this possible or are you not interested in this at all for now?

@bdekonin it's not something we have enough bandwidth on our team for at the moment.

**Comment by haarchri** [comment]
I have also a bunch of cloudformation resources in the pipeline ;(

**Comment by NickDeitmers** [comment]
Same for me, I think it would be a great feature to have cloudformation resources supported! I was planning to contribute on this as well.

File: infracost-usage-example.yml [hunk 1]
```diff
@@ -9,6 +9,17 @@ resource_type_default_usage:
     request_duration_ms: 500 # Average duration of each request in milliseconds.
 resource_usage:
   #
+  # Cloudformation AWS resources
```
Comment (aliscott) [inline infracost-usage-example.yml:12]: I'm going to remove this example example for just now. We're currently reworking some of the usage stuff. If we have examples for CloudFormation in the future we should keep them separate from the Terraform ones.

**Review by aliscott — APPROVED** [review]
Thanks @haarchri. I'll rebase and merge this.

**Pull request merged**